src/TypeCollector.php

```
@@ -524,87 +519,64 @@ private function castValueToArray(mixed $value, string $separator): array
+                } else {
+                    $typeName = $type->getName();
 
-        foreach ($this->reflection->getProperties() as $property) {
-            if ($property->isStatic()) {
-                continue;
+                    if ($type->allowsNull() && $typeName !== 'null') {
+                        $this->properties[$propertyName] = [$typeName, 'null'];
+                    } else {
+                        $this->properties[$propertyName] = $typeName;
+                    }
+                }
```

PHPMD (2.15.0)

[error] The method collectMetadata uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)